umm-lib/src/cmr/umm/collection/product_specific_attribute.clj

```diff
@@ -61,26 +34,27 @@
 (defn find-formatter
   [datetime regex-formatter-map]
   (->> regex-formatter-map
-       (filter (fn [[regex formatter]]
+       (filter (fn [[regex _formatter]]
                  (re-matches regex datetime)))
        first
        second))
 
-(defmethod parse-value :datetime
-  [data-type value]
-  (when value
-    (f/parse (find-formatter value datetime-regex->formatter) value)))
-
-(defmethod parse-value :time
-  [data-type value]
-  (when value
-    (f/parse (find-formatter value time-regex->formatter) value)))
-
-(defmethod parse-value :date
+(defn parse-value
+  "Parses a value based on the data type given"
   [data-type value]
   (when value
-    (let [value (str/replace value "Z" "")]
-      (f/parse (f/formatters :date) value))))
+    (case data-type
```

Comment:
I might like this linter more now

Reply:
The linter doesn't point this out.  I learned from Duncan and from reading that polymorphism is slow in clojure and most of the time the way we use it, it is not necessary.  I have been changing the defmulti to case or cond when ever I refactor code when I can. I have been doing this way before this PI started.  While I like to have some time to clean up code, this linting stuff has been a real waste of time in my opinion.